### PR TITLE
Sidebar menu: Register "plugin install" route on sites with the default interface

### PR DESCRIPTION
**Pull request by mmtr**
## Proposed changes:

In https://github.com/Automattic/jetpack/pull/37521 we added a new "Add New Plugin" submenu which is only registered on sites with the classic admin interface so we could redirect from `/wp-admin/plugin-install.php` to the route of that submenu (`/wp-admin/plugins.php?page=wpcom-install-plugin`).

However, for that redirect to work, we need the new route to be working on all Simple sites, including sites with the default admin interface (see pMz3w-k1d-p2#comment-119194).

Since we don't want the "Add New Plugin" submenu to be visible on sites with the default admin interface, this PR implements a workaround in which it registers the menu (on all sites) and then hides it (on sites with the default interface).

To simplify the workaround, the existing logic that registers the Plugins parent menu from Nav Unification to the jetpack-mu-wpcom package.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Apply these changes to your WP.com simple site
- With the classic admin interface, make sure you get the same behavior as before (3 submenus: Plugins > Installed Plugins, Plugins > Add New Plugin, Plugins > Marketplace)
- With the default admin interface:
  - Make sure Plugins still links to the Marketplace
  - Make sure there are no submenus
  - Make sure you can visit `/wp-admin/plugins.php?page=wpcom-install-plugin`

### CODE REVIEW

File: projects/packages/masterbar/src/admin-menu/class-wpcom-admin-menu.php [hunk 1]
```diff
@@ -411,38 +411,6 @@ public function should_link_to_wp_admin() {
...
-		// TODO: Remove wpcom_menu (/wp-content/admin-plugins/wpcom-misc.php).
-		$count = '';
-		if ( ! is_multisite() && current_user_can( 'update_plugins' ) ) {
-			$update_data = wp_get_update_data();
-			$count       = sprintf(
-				'<span class="update-plugins count-%s"><span class="plugin-count">%s</span></span>',
-				$update_data['counts']['plugins'],
-				number_format_i18n( $update_data['counts']['plugins'] )
-			);
-		}
```
Comment (mmtr) [inline projects/packages/masterbar/src/admin-menu/class-wpcom-admin-menu.php:427]: This is some logic that we copy/pasted from a file that was shared between Simple and Atomic sites. For Simple sites, we don't need this logic (`is_multisite` always returns true), which is why I didn't incorporated into the jetpack-mu-wpcom package.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php [hunk 2]
```diff
@@ -395,26 +395,6 @@ public function should_link_to_wp_admin() {
...
-		// TODO: Remove wpcom_menu (/wp-content/admin-plugins/wpcom-misc.php).
-		$count = '';
-		if ( ! is_multisite() && current_user_can( 'update_plugins' ) ) {
-			$update_data = wp_get_update_data();
-			$count       = sprintf(
-				'<span class="update-plugins count-%s"><span class="plugin-count">%s</span></span>',
-				$update_data['counts']['plugins'],
-				number_format_i18n( $update_data['counts']['plugins'] )
-			);
-		}
```
Comment (mmtr) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php:411]: This is some logic that we copy/pasted from a file that was shared between Simple and Atomic sites. For Simple sites, we don't need this logic (`is_multisite` always returns true), which is why I didn't incorporated into the jetpack-mu-wpcom package.

**Comment by rcrdortiz** [comment]
> Since we don't want the "Add New Plugin" submenu to be visible on sites with the default admin interface, this PR implements a workaround in which it registers the menu (on all sites) and then hides it (on sites with the default interface).

Are we 100% sure about this? What are the cons and pros of hiding the add new plugin page on the default admin interface?

**Review by dsas — APPROVED** [review]
Works as described, lets answer Richard's question before merging though.

**Comment by rcrdortiz** [comment]
Would it make sense to highlight the `Plugins` menu on sites that use the Default interface?

Screenshot 2024-06-04 at 13 13 35

With the current implementation, it looks like this:

Screenshot 2024-06-04 at 13 14 43

**Review by rcrdortiz — APPROVED** [review]
Changes work as described.

**Comment by mmtr** [comment]
> Are we 100% sure about this? What are the cons and pros of hiding the add new plugin page on the default admin interface?

Not 100% sure, but it is what we decided when we launched Nav Unification a few years ago, so I just wanted to avoid revisiting that decision.

> Would it make sense to highlight the Plugins menu on sites that use the Default interface?

It makes total sense, but unfortunately it's not-so-trivial technically if the submenu remains hidden.

**Pull request merged**